src/topology/uniform_space/basic.lean

```diff
@@ -1384,6 +1436,48 @@ lemma to_topological_space_prod {α} {β} [u : uniform_space α] [v : uniform_sp
   @uniform_space.to_topological_space (α × β) prod.uniform_space =
     @prod.topological_space α β u.to_topological_space v.to_topological_space := rfl
 
+/-- A version of `uniform_continuous_inf_dom_left` for binary functions -/
+lemma uniform_continuous_inf_dom_left₂ {α β γ} {f : α → β → γ}
+  {ua1 ua2 : uniform_space α} {ub1 ub2 : uniform_space β} {uc1 : uniform_space γ}
+  (h : by haveI := ua1; haveI := ub1; exact uniform_continuous (λ p : α × β, f p.1 p.2)) :
+  by haveI := ua1 ⊓ ua2; haveI := ub1 ⊓ ub2; exact uniform_continuous (λ p : α × β, f p.1 p.2) :=
```

Comment:
Well that would definitely help over expanding everything with one single `@`, but that doesn't solve the main problem because I would need to invoke explicitly `prod.uniform_space`

Reply:
I forget that you're basically copying my code here, so I must already have come to that conclusion myself!